templates/404.html

```diff
@@ -4,13 +4,13 @@
 <div class="p-strip--light">
   <div class="row">
     <h1>404: Sorry, page not found.</h1>
-    <p>Try a different URL, try searching for solutions or learn how to <a class="p-link--external" href="https://docs.jujucharms.com/authors-charm-writing">create your own solution</a>.</p>
+    <p>Try a different URL, try searching for solutions or learn how to <a class="p-link--external" href="{{ external_urls.docs }}authors-charm-writing">create your own solution</a>.</p>
     <nav>
       <ul class="p-inline-list">
-        <li class="p-inline-list__item"><a href="/store" class="p-button--positive">Browse the store</a></li>
-        <li class="p-inline-list__item"><a href="https://jujucharms.com/q/?type=bundle">All bundles</a></li>
-        <li class="p-inline-list__item"><a href="https://jujucharms.com/q/?type=charm">All charms</a></li>
-        <li class="p-inline-list__item"><a href="https://github.com/CanonicalLtd/jujucharms.com/issues" class="p-link--external">Submit a bug</a></li>
+        <li class="p-inline-list__item"><a href="{{ url_for('jaasstore.store') }}" class="p-button--positive">Browse the store</a></li>
+        <li class="p-inline-list__item"><a href="{{ url_for('jaasstore.search') }}?type=bundle">All bundles</a></li>
+        <li class="p-inline-list__item"><a href="{{ url_for('jaasstore.search') }}?type=charm">All charms</a></li>
```

Comment:
Why do some of these use `url_for` and others use `external_urls` ?

If this is generating a url, should it not also take the query params as arguments?